src/docs/blueprint_documentation.md

```diff
@@ -0,0 +1,180 @@
+
+To install dependencies and build the application, simply execute `yarn`.
+
+You can also directly run the unpackaged application, e.g. to test it during development with `yarn start`.
+
+With `yarn package`, you package the application into an executable file for your current operating system.
+The packaged application will be located in `electron-app/dist`.
+The folder `electron-app/dist/<OS>-unpackaged` will contain the files that are bundled into the final packaged executable.
+For Linux, this is an executable `.AppImage`, for Windows a `.exe` installer, and a `.pkg` installer for macOS.
+
+You can also just create the unpackaged content by running `yarn package:preview`.
+This is useful to see the bundled files and saves time compared to a full package.
+To publish the current version, the command `yarn deploy` can be used.
+For more information on publishing also see section "Configure publish and update".
+
+## Signing
+
+[Electron-builder](https://www.electron.build/) supports signing the packaged application on Windows and macOS.
+The current signing scripts for Theia Blueprint are located in `electron-app/scripts`.
+The file `after-pack.js` is the current entry point for the configured signing via Eclipse infrastructure.
+
+However, as signing is highly dependent on your setup, see the [electron builder’s signing documentation](https://www.electron.build/code-signing) on how to properly set up your own signing.
+
+## Updating Bundled VS Code Extensions
+
+All VS Code extensions that are already included in the product at start-up, are defined in `electron-app/package.json`.
+They are listed under the `theiaPlugins` property as key-value pairs.
+The keys can be freely chosen as long as they represent a valid folder name and are unique within the `theiaPlugins` property.
+We suggest using the extension’s unique identifier.
+The value is the download URL of the extensions.
+It will automatically be downloaded during the application’s build process.
+Any new plugin will be automatically downloaded the next time one of the following npm scripts is executed:
+
+- `install` (which is the same as just running `yarn`)
+- `prepare`
+- `download:plugins`
+
+To remove an extension from the product, simply delete its entry.
+If plugins were not already downloaded, no further steps are required as downloaded plugins are ignored via gitignore.
+However, previously downloaded plugins are not automatically removed.
+Therefore, you need to remove its folder from the `electron-app/plugins` folder.
+Alternatively, you can remove the whole `electron-app/plugins` folder and execute `yarn run download:plugins` to download all defined plugins.
+
+### Extension sources
+
+While you can technically install extensions directly from [Microsoft’s VS Code Marketplace](https://marketplace.visualstudio.com/), you should check the Terms of Use to ensure compliance as Microsoft typically restricts usage to only VS Code.
```

Comment:
I think we can omit this line, we would not want to encourage usage of the ms marketplace which is against their terms of use. In addition, "install extensions" is a little too generic, are you referring to extensions bundled with the application through the `package.json`, or through the extensions-view? (if through the extensions-view we do not support ms marketplace installations).